plugins/format/plugin.js

```diff
@@ -64,10 +64,10 @@ CKEDITOR.plugins.add( 'format', {
 				editor.focus();
 				editor.fire( 'saveSnapshot' );
 
-				var style = styles[ value ],
-					elementPath = editor.elementPath();
+				var style = styles[ value ];
 
-				editor[ style.checkActive( elementPath, editor ) ? 'removeStyle' : 'applyStyle' ]( style );
+				// Always aply style, do not allow on toggle it by clicking on list item (#584).
+				editor.applyStyle( style );
```

Comment:
By default all format tags are block elements so reapplying them to partial selection doesn't do anything. However, if you change the config like:

```
config.format_p = { element: 'span', attributes: { 'class': 'normalPara' } };
```

You can apply it to partial selection despite the fact it was already applied (so it creates nested tags). This is an edge case IMHO but still we should handle this. You could just check if style is active (as before `style.checkActive( elementPath, editor )`) and apply it only if it is not already there.